azurerm/internal/common/client_options.go

```diff
@@ -63,8 +63,14 @@ func setUserAgent(client *autorest.Client, tfVersion, partnerID string) {
 		client.UserAgent = fmt.Sprintf("%s %s", client.UserAgent, azureAgent)
 	}
 
+	// only one pid can be interpreted currently
+	//  hence, send partner Id if present
 	if partnerID != "" {
 		client.UserAgent = fmt.Sprintf("%s pid-%s", client.UserAgent, partnerID)
+	} else {
+		// otherwise, if pid is not present in config files, send Terraform GUID
+		terraformGUID := "222c6c49-1b0a-5959-a213-6608f9eb8820"
+		client.UserAgent = fmt.Sprintf("%s pid-%s", client.UserAgent, terraformGUID)
```

Comment:
Why not use only a single Sprintf after the if else (with a local variable), to  eliminate the code duplication?

Reply:
```suggestion
if partnerID = "" {
	partnerID := "222c6c49-1b0a-5959-a213-6608f9eb8820"
}
client.UserAgent = fmt.Sprintf("%s pid-%s", client.UserAgent, partnerID)
```